### PR TITLE
Quick Start: New visuals for Quick Start Modal

### LINKED ISSUE #16588: Quick Start: New visuals for checklist modal
Parent #16347

## Description

Visual changes to the Quick Start checklist modal.

Design ref: pbArwn-4wS-p2

## Tasks

- [x] Update Quick Start checklist modal 
    - [x] #16611
    - [x] #16622
    - [x] #16639

### PR DESCRIPTION
**Pull request by ashiagr**
Closes #16588

This PR applies new visuals to the quick start modal.

Type  | Light Mode | Dark Mode
-|-|-
Know the app tour | get_to_know_app_light| get_to_know_app_dark
Customize tour | customize_your_site_light | customize_your_site_dark
Grow tour | grow_your_audience_light | grow_your_audience_dark

RTL
---
 

Jetpack
---

## Testing Instructions

_Please test the following using both quick start types and on light and dark modes_
1. Enable quick start
2. Tap on the quick start card
3. Make sure the modal is presented and matches the design
4. Tap any task
5. Make sure the modal is dismissed and the task is started
6. Finish the task
7. Open the modal again
8. Make sure the completed task matches the design

### Notes:

1. `Edit your homepage` task is not yet removed in `Android` in the `Customize your site` tour. Chosen blue color same as used for social sharing in `Grow your audience` tour.
2. `grey 800` used for card background in dark mode is not present in the [palette grey colors](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/res/values/colors_studio_palette.xml#L10-L22)). Also `grey 100` is not `f5f5f5` but a different shade of grey in the palette. I used colors that closely match the given colors. 
`gray_0` - light mode
`on surface (alpha 0.12)` - dark mode
3. Known Issue: In Landscape mode layout, title is set above the tasks but not on the app bar. Setting it on the app bar, with custom font, centering it and toggling it on configuration change requires changes to the base dialog fragment used across the app and will be taken up separately.
4. Known Issue: Loading indicator not going away on my site screen - it is not related to changes in this PR. I'll look into it separately. - Fixed in #16646

## Regression Notes
1. Potential unintended areas of impact
Any crashes/ unusual layout on lower API level due to unsupported API. 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested manually on min supported API level 24 emulator. 

5. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary. - To be added after new visuals are applied to the quick start card on my site screen.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/quickstart/QuickStartFullScreenDialogFragment.kt [hunk 1]
```diff
@@ -72,7 +82,17 @@ class QuickStartFullScreenDialogFragment : Fragment(R.layout.quick_start_dialog_
...
+            add(
+                    QuickStartHeaderCard(
+                            title = UiStringRes(headerTitleResId),
+                            shouldShowHeaderImage = !displayUtilsWrapper.isPhoneLandscape()
+                    )
+            )
+            addAll(taskCards)
+        }.toList()
+        quickStartAdapter.submitList(quickStartList)
     }
```
Comment (AjeshRPai) [inline WordPress/src/main/java/org/wordpress/android/ui/quickstart/QuickStartFullScreenDialogFragment.kt:87]: Suggestion: I think we can extract the lines 85-94 to a function which creates and returns a `QuickStartHeaderCard` and then add it to the list. WDYT 🤔 . Something like this 

```
fun getHeaderCard() {
        val headerTitleResId = quickStartCardBuilder.getTitle(tasksType)
        val header =  QuickStartHeaderCard(
                            title = UiStringRes(headerTitleResId),
                            shouldShowHeaderImage = !displayUtilsWrapper.isPhoneLandscape()
          )
       return header
}
```

**Review by AjeshRPai — APPROVED** [review]
I have looked at the code changes and tested it as per the instructions. Everything looks good to me. @ashiagr. 👍 🙌 

I have left a suggestion for you to address, if you feel that change is not necessary please feel free to merge. As that change is non blocking, I am approving the PR

File: WordPress/src/main/java/org/wordpress/android/ui/quickstart/QuickStartFullScreenDialogFragment.kt (hunk 1 above)
Comment (ashiagr) [reply WordPress/src/main/java/org/wordpress/android/ui/quickstart/QuickStartFullScreenDialogFragment.kt:87]: Good point.

Actually, this fragment needs to utilize `ViewModel` and Card-`Builder` pattern, hopefully, we'll get time to do that in near future. Till that time, included your suggestion in 8e6c6b0.

**Review by osullivanchris — APPROVED** [review]
This is all looking pretty tight to me @ashiagr ! Think I can approve with no nitpicks / tweaks, nice job 😄

File: WordPress/src/main/java/org/wordpress/android/ui/quickstart/QuickStartFullScreenDialogFragment.kt (hunk 1 above)
Comment (AjeshRPai) [reply WordPress/src/main/java/org/wordpress/android/ui/quickstart/QuickStartFullScreenDialogFragment.kt:87]: Yep, Thanks for making the change. 👍🏼

**Pull request merged**